http/src/main/java/io/hyperfoil/http/connection/HttpChannelInitializer.java

```diff
@@ -71,6 +72,11 @@ protected void initChannel(Channel ch) {
          }
          long sslHandshakeTimeout = clientPool.config().sslHandshakeTimeout();
          sslHandler.setHandshakeTimeoutMillis(sslHandshakeTimeout < 0 ? 0 : sslHandshakeTimeout);
+         sslHandler.handshakeFuture().addListener(future -> {
+            if (!future.isSuccess() && future.cause() instanceof SslHandshakeTimeoutException) {
```

Comment:
maybe handle other failures too

Reply:
yeah I think that's safe to remove the instanceOf check and handle all possible issues. Will do